vsphere/datadog_checks/vsphere/api_rest.py

```diff
@@ -42,9 +51,23 @@ def smart_connect(self):
         """
         self._client.connect_session()
 
-    @smart_retry
-    def get_resource_tags(self):
-        # type: () -> ResourceTags
+    def make_batch(self, mors):
+        # type: (Iterator[vim.ManagedEntity]) -> Iterator[List[vim.ManagedEntity]]
+        batch = []  # type: List[vim.ManagedEntity]
+        size = self.config.batch_tags_collector_size
+        for mor in mors:
+            if len(batch) == size:
+                yield batch
+                batch = []
+                continue
+            batch.append(mor)
+        else:
+            if batch:
+                yield batch
```

Comment:
Maybe we can add something like this as util function in base instead? https://stackoverflow.com/a/434328

Reply:
The code you've linked only work on lists, not on iterator

Hard to match all needs:
- what if the input is a list/iterator
- what if the output expects padding
- what if the condition is trickier than just the batch size (the other place where we make batch as a more complex condition).
- what if you need the output to be an iterator?

Also I don't think we have other use cases for that at the moment.

Reply:
Sure. Maybe YAGNI. 

(minor) I think it worth add few unit tests for `make_batch`